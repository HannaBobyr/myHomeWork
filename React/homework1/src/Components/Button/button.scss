.button {
  border-width: 0.5px;
  padding: 8px;
  border-radius: 5px;
  cursor: pointer;
  color: white;
  margin: 10px;
  &--close {
      position: absolute;
      right: 0px;
      top: 0px;
      border: none;
    }
}
