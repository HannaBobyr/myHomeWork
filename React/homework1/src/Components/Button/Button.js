import React, { Component } from "react";
import "./button.scss";
class Button extends Component {
  render() {
    const {
      text,
      backgroundColor,
      onClick,
      className,
      isActiveShare,
      isActiveDelete,
    } = this.props;
    return (
      !(isActiveShare || isActiveDelete) && (
        <button
          className={className}
          type="button"
          onClick={onClick}
          style={{ backgroundColor: `${backgroundColor}` }}
        >
          {text}
        </button>
      )
    );
  }
}
export default Button;
