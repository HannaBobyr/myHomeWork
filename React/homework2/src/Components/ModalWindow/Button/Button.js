import React, { Component } from "react";
import "./button.scss";
import PropTypes from "prop-types";
class Button extends Component {
  render() {
    const { text, onClick, className, backgroundColor } = this.props;
    return (
      <button
        className={className}
        type="button"
        onClick={onClick}
        style={{ backgroundColor: `${backgroundColor}` }}
      >
        {text}
      </button>
    );
  }
}
export default Button;
Button.propTypes = {
  className: PropTypes.string,
  text: PropTypes.string,
  backgroundColor: PropTypes.string,
  onClick: PropTypes.func,
};
