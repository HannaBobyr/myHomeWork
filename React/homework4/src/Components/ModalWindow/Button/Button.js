import React from "react";
import "./button.scss";
import PropTypes from "prop-types";
function Button({ text, onClick, className, backgroundColor }) {
  return (
    <button
      className={className}
      type="button"
      onClick={onClick}
      style={{ backgroundColor: `${backgroundColor}` }}
    >
      {text}
    </button>
  );
}

export default Button;
Button.propTypes = {
  className: PropTypes.string,
  text: PropTypes.string,
  backgroundColor: PropTypes.string,
  onClick: PropTypes.func,
};
