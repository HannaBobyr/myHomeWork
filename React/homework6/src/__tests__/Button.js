import { render, screen } from "@testing-library/react";
import Button from "Components/ModalWindow/Button/Button";
import renderer from "react-test-renderer";
describe("Test Buttons", () => {
  it("render button", () => {
    render(
      <Button
        className="button"
        text="OK"
        backgroundColor="rgb(90, 178, 255)"
      />
    );
    expect(screen.getByRole("button")).toBeInTheDocument();
  });
  it("button have class", () => {
    render(
      <Button
        className="button"
        text="OK"
        backgroundColor="rgb(90, 178, 255)"
      />
    );
    const btn = screen.getByRole("button");
    expect(btn).toHaveClass("button");
  });
  it("button have text", () => {
    render(
      <Button
        className="button"
        text="OK"
        backgroundColor="rgb(90, 178, 255)"
      />
    );
    const btn = screen.getByRole("button");
    expect(btn).toHaveTextContent(/ok/i);
  });
  it("Matches snapshot", () => {
    const tree = renderer
      .create(
        <Button
          className="button"
          text="OK"
          backgroundColor="rgb(90, 178, 255)"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
