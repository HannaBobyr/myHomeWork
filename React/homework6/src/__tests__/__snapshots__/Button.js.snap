// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Buttons Matches snapshot 1`] = `
<button
  className="button"
  data-testid="button"
  style={
    Object {
      "backgroundColor": "rgb(90, 178, 255)",
    }
  }
  type="button"
>
  OK
</button>
`;
